FileShareClient/FileShareApiClient.cs

```diff
@@ -76,8 +81,8 @@ public async Task<BatchSearchResponse> Search(string searchQuery, int? pageSize
 
             using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, uri))
             {
-                var response = await httpClientFactory.CreateClient()
-                    .SendAsync(httpRequestMessage, CancellationToken.None);
+                var httpClient = await httpClientFactory.CreateClient().SetAuthenticationHeader(authTokenProvider);
```

Comment:
This line (`await httpClientFactory.CreateClient().SetAuthenticationHeader(authTokenProvider)`) is repeated many time throughout the class. Could it be collapsed to a single private (protected?) method call?  e.g.
```cs
protected Task<IHttpClient> GetAuthenticatedClient() => httpClientFactory.CreateClient().SetAuthenticationHeader(authTokenProvider);
```